corehq/form_processor/submission_post.py

```diff
@@ -222,15 +224,35 @@ def run(self):
                         raise
                     else:
                         instance.initial_processing_complete = True
-                        error_message = self.save_processed_models(case_db, xforms, case_stock_result)
-                        if error_message:
-                            response_nature = ResponseNature.POST_PROCESSING_FAILURE
+                        openrosa_kwargs['error_message'] = self.save_processed_models(case_db, xforms,
+                                                                                      case_stock_result)
+                        if openrosa_kwargs['error_message']:
+                            openrosa_kwargs['error_nature'] = ResponseNature.POST_PROCESSING_FAILURE
                         cases = case_stock_result.case_models
                         ledgers = case_stock_result.stock_result.models_to_save
+
+                        from corehq.apps.export.views import CaseExportListView, FormExportListView
+                        from corehq.apps.reports.views import CaseDetailsView, FormDataView
+                        messages = [_('Form successfully saved!')]
+                        form_link = reverse(FormDataView.urlname, args=[instance.domain, instance._id])
+                        form_export_link = reverse(FormExportListView.urlname, args=[instance.domain])
+                        if cases:
+                            case_link = reverse(CaseDetailsView.urlname, args=[instance.domain, cases[0]._id])
+                            case_export_link = reverse(CaseExportListView.urlname, args=[instance.domain])
+                            messages.append(
+                                _("You just submitted [this form]({}), which affected [this case]({}).")
+                                .format(form_link, case_link))
+                            messages.append(
+                                _("Click to export your [case]({}) or [form]({}) data.")
+                                .format(case_export_link, form_export_link))
+                        else:
+                            messages.append(_("You just submitted [this form]({}).").format(form_link))
+                            messages.append(_("Click to export your [form]({}) data.").format(form_export_link))
+                        openrosa_kwargs['success_message'] = "\n\n".join(messages)
```

Comment:
I don't think we want to include these messages for normal web apps submissions or for mobile submissions